Code/Framework/AzToolsFramework/AzToolsFramework/Asset/AssetDebugInfo.cpp

```diff
@@ -37,36 +37,58 @@ namespace AzToolsFramework
             //!   fonts/vera.fontfamily - {6B62FA48-7032-5B8D-88DB-4EDCF0D500AE}:0
             //!     ui/canvas/start.uicanvas - {E7B190F9-3507-5524-BE05-35C6941B8E26}:0
             //!       [SEED] levels/testdependencieslevel/level.pak - {A89A2DCA-0C7B-5919-B3E0-3F67BFD8AA40}:0
-        void BuildHumanReadableString(const AZStd::string& tabString, AssetFileDebugInfoList& infoList)
+        void BuildHumanReadableString(const AZStd::string& tabString, AssetFileDebugInfoList& infoList, AZStd::string& outputString, bool bLogAsErrors = false)
         {
             AZStd::string fileName = infoList.m_fileDebugInfoList[m_assetId].m_assetRelativePath;
 
             // Only print size on the top asset, not the graph below it.
             uint64_t filesize = infoList.m_fileDebugInfoList[m_assetId].m_fileSize;
             AZStd::string sizeString = tabString.empty() ? AZStd::string::format(" - %" PRIu64 " bytes", filesize) : "";
 
-            infoList.m_humanReadableString += tabString;
+            outputString += tabString;
 
             if (m_isCyclicalDependency)
             {
-                infoList.m_humanReadableString += AZStd::string("[CYCLICAL DEPENDENCY] ");
+                outputString += AZStd::string("[CYCLICAL DEPENDENCY] ");
             }
             else if (m_leaves.empty())
             {
-                infoList.m_humanReadableString += AZStd::string("[SEED] ");
+                outputString += AZStd::string("[SEED] ");
             }
 
-            infoList.m_humanReadableString += AZStd::string::format("%s - %s%s\n",
+            outputString += AZStd::string::format("%s - %s%s\n",
                 fileName.c_str(),
                 m_assetId.ToString<AZStd::string>().c_str(),
                 sizeString.c_str());
 
+            if (bLogAsErrors)
+            {
+                AZStd::string errorText;
+
+                if (m_isCyclicalDependency)
+                {
+                    errorText += AZStd::string("[CYCLICAL DEPENDENCY] ");
+                }
+                else if (m_leaves.empty())
+                {
+                    errorText += AZStd::string("[SEED] ");
+                }
+
+               errorText += AZStd::string::format("%s - %s%s",
+                    fileName.c_str(),
+                    m_assetId.ToString<AZStd::string>().c_str(),
+                    sizeString.c_str());
+
+               AZ_Error("Dependency", false, errorText.c_str());
```

Comment:
This opens up the possibility of format string injection if the user provides a file name with a "%n" in it's name.
i.e "C:/O3DE/MyAssetFile%nFile".

The errorText.c_str() should be not be used directly as the format string, but formated into a `%s` argument
```suggestion
               errorText += AZStd::string::format("%s - %s%s",
                    fileName.c_str(),
                    m_assetId.ToString<AZStd::string>().c_str(),
                    sizeString.c_str());

               AZ_Error("Dependency", false, "%s", errorText.c_str());
```